packages-ts/gauntlet-terra-cw20-multisig/src/commands/multisig.ts

```diff
@@ -0,0 +1,275 @@
+      const executeInput = {
+        execute: {
+          proposal_id: proposalId,
+        },
+      }
+      return new MsgExecuteContract(signer, this.multisig, executeInput)
+    }
+
+    fetchState = async (proposalId?: number): Promise<State> => {
+      const groupState = await this.query(this.multisigGroup, {
+        list_members: {},
+      })
+      const owners = groupState.members.map((m) => m.addr)
+      const thresholdState = await this.query(this.multisig, {
+        threshold: {},
+      })
+      const threshold = thresholdState.absolute_count.total_weight
+      if (!proposalId) {
+        return {
+          threshold,
+          nextAction: Action.CREATE,
+          owners,
+          approvers: [],
+        }
+      }
+      const proposalState = await this.query(this.multisig, {
+        proposal: {
+          proposal_id: proposalId,
+        },
+      })
+      const votes = await this.query(this.multisig, {
+        list_votes: {
+          proposal_id: proposalId,
+        },
+      })
+      const status = proposalState.status
+      const toNextAction = {
+        passed: Action.EXECUTE,
+        open: Action.APPROVE,
+        pending: Action.APPROVE,
+        rejected: Action.NONE,
+        executed: Action.NONE,
+      }
+      return {
+        threshold,
+        nextAction: toNextAction[status],
+        owners,
+        currentStatus: status,
+        data: proposalState.msgs,
+        approvers: votes.votes.filter((v) => v.vote === Vote.YES).map((v) => v.voter),
+      }
+    }
+
+    printPostInstructions = async (proposalId: number) => {
+      const state = await this.fetchState(proposalId)
+      const approvalsLeft = state.threshold - state.approvers.length
+      const messages = {
+        [Action.APPROVE]: `The proposal needs ${approvalsLeft} more approvals. Run the same command with the flag --proposal=${proposalId}`,
+        [Action.EXECUTE]: `The proposal reached the threshold and can be executed. Run the same command with the flag --proposal=${proposalId}`,
+        [Action.NONE]: `The proposal has been executed. No more actions needed`,
+      }
+      logger.line()
+      logger.info(`${messages[state.nextAction]}`)
+      logger.line()
+    }
+
+    execute = async () => {
+      let proposalId = !!this.flags.proposal && Number(this.flags.proposal)
+      const state = await this.fetchState(proposalId)
+
+      if (state.nextAction === Action.NONE) {
+        await this.printPostInstructions(proposalId)
+        return
+      }
+      const rawTx = await this.makeRawTransaction(this.wallet.key.accAddress, state)
+
+      logger.info(`Proposal State:
+        - Total Owners: ${state.owners.length}
+        - Owners List: ${state.owners}
+
+        - Threshold: ${state.threshold}
+        - Total Approvers: ${state.approvers.length}
+        - Approvers List: ${state.approvers}
+
+        - Next Action: ${state.nextAction.toUpperCase()}
+      `)
```

Comment:
This is another inspection command baked in ad-hoc.

Reply:
#133